examples/06_material/main.rs

```diff
@@ -0,0 +1,182 @@
+                let roughness = Texture::from_color_val([roughness, roughness, roughness, 1.0]);
+
+                let mtl = mtl.clone().with_metallic(metallic).with_roughness(roughness);
+
+                let mtl = load_proc_asset(engine, move |engine| {
+                    let factory = engine.world.read_resource::<Factory>();
+                    factory
+                        .create_material(mtl)
+                        .map(MaterialComponent)
+                        .map_err(BoxedErr::new)
+                });
+                engine
+                    .world
+                    .create_entity()
+                    .with(Transform(pos.into()))
+                    .with(mesh.clone())
+                    .with(mtl)
+                    .build();
+            }
+        }
+
+
+        println!("Create lights");
+        engine
+            .world
+            .create_entity()
+            .with(LightComponent(
+                PointLight {
+                    center: [6.0, 6.0, -6.0].into(),
+                    intensity: 6.0,
+                    color: [0.8, 0.0, 0.0].into(),
+                    ..PointLight::default()
+                }.into(),
+            ))
+            .build();
+
+        engine
+            .world
+            .create_entity()
+            .with(LightComponent(
+                PointLight {
+                    center: [6.0, -6.0, -6.0].into(),
+                    intensity: 5.0,
+                    color: [0.0, 0.3, 0.7].into(),
+                    ..PointLight::default()
+                }.into(),
+            ))
+            .build();
+
+        println!("Put camera");
+        engine.world.add_resource(Camera {
+            eye: [0.0, 0.0, -12.0].into(),
+            proj: Projection::perspective(1.3, Deg(60.0)).into(),
+            forward: [0.0, 0.0, 1.0].into(),
+            right: [1.0, 0.0, 0.0].into(),
+            up: [0.0, 1.0, 0.0].into(),
+        });
+    }
+
+    fn handle_event(&mut self, _: &mut Engine, event: Event) -> Trans {
+        match event {
+            Event::WindowEvent { event, .. } => {
+                match event {
+                    WindowEvent::KeyboardInput {
+                        input: KeyboardInput { virtual_keycode: Some(VirtualKeyCode::Escape), .. }, ..
+                    } |
+                    WindowEvent::Closed => Trans::Quit,
+                    _ => Trans::None,
+                }
+            }
+            _ => Trans::None,
+        }
+    }
+}
+
+
+type DrawPbm = pass::DrawPbm<PosNormTangTex, AmbientColor, MeshComponent, MaterialComponent, Transform, LightComponent>;
+
+fn run() -> Result<(), amethyst::Error> {
+    let path = format!(
+        "{}/examples/06_material/resources/config.ron",
+        env!("CARGO_MANIFEST_DIR")
+    );
+    let config = DisplayConfig::load(&path);
+    let mut game = Application::build(Example)?
+        .with_renderer(
+            ThePipeline::build().with_stage(
```

Comment:
I'm not sure we want a type named `ThePipeline`

Reply:
Just because `Pipeline` is occupied by trait 😛

Reply:
I guess my question is why do we have a trait for this?  Under what circumstances would someone want to use a pipeline other than ours?

Reply:
We have a trait to simplify bounds on type parameters.
Either it is `P where P: Pipeline` or `Pipeline<S> where S: Stages`.
Well. Before refactoring the second variant was much more sophisticated...

Reply:
And I think that `PIpeline<S> where S: Stages` looks unclear rather than `P: Pipeline`.